include/myst/thread.h

```diff
@@ -405,13 +438,66 @@ MYST_INLINE int myst_thread_queue_remove_thread(
     return found ? pos : -1;
 }
 
+/** Find and remove up to `n` threads from the queue such that
+ * <waiting bitset> & <passed in bitset> != 0.
+ * Removed threads are placed in the `matches` queue.
+ */
+MYST_INLINE size_t myst_thread_queue_search_remove_bitset(
```

Comment:
use return type `int` given you're returning -1 on failure

Reply:
Can you elaborate on the reasoning for this? I based this function off of `myst_thread_queue_remove_thread()`.
![image](https://user-images.githubusercontent.com/9376183/132071211-c0770ef1-f3d2-4612-8403-13a93006b011.png)

Reply:
The reference you put uses int

Reply:
Got it, I initially misread your comment as removing the MYST_INLINE